akka-persistence-typed/src/main/scala/akka/persistence/typed/EventSourcedSignal.scala

```diff
@@ -29,6 +31,56 @@ final case class RecoveryFailed(failure: Throwable) extends EventSourcedSignal {
   def getFailure(): Throwable = failure
 }
 
+final case class PersistFailed[Command, Event](failure: Throwable, command: Option[Command], event: Event)
+    extends EventSourcedSignal {
+
+  /**
+   * Java API
+   */
+  def getFailure(): Throwable = failure
+
+  /**
+   * Java API
+   */
+  def getCommand(): Optional[Command] = {
+    import scala.compat.java8.OptionConverters._
+    command.asJava
+  }
+
+  /**
+   * Java API
+   */
+  def getEvent(): Event = event
```

Comment:
Ok, that explains why `WriteMessageRejected` can only have a single event. 

If we can't return the list of events that the user tried to persist, then indeed better to not pass any. If can be confusing as it might give the impression that only that event was not persisted.

Reply:
I removed the event parameter